pkg/cloud/azure/azure_test.go

```diff
@@ -79,3 +91,173 @@ func TestResourcesRenderingSmoke(t *testing.T) {
+			infra:    makeInfrastructureResource(configv1.AzurePlatformType, configv1.AzurePublicCloud),
+		},
+		{
+			name: "Azure sets the cloud to AzurePublicCloud and keeps existing fields",
+			source: azure.Config{
+				ResourceGroup: "test-rg",
+			},
+			expected: azure.Config{VMType: "standard", ResourceGroup: "test-rg", AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzurePublicCloud)}},
+			infra:    makeInfrastructureResource(configv1.AzurePlatformType, configv1.AzurePublicCloud),
+		},
+		{
+			name:     "Azure keeps the cloud set to AzurePublicCloud",
+			source:   azure.Config{AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzurePublicCloud)}},
+			expected: azure.Config{VMType: "standard", AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzurePublicCloud)}},
+			infra:    makeInfrastructureResource(configv1.AzurePlatformType, configv1.AzurePublicCloud),
+		},
+		{
+			name:     "Azure keeps the cloud set to US Gov cloud",
+			source:   azure.Config{AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzureUSGovernmentCloud)}},
+			expected: azure.Config{VMType: "standard", AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzureUSGovernmentCloud)}},
+			infra:    makeInfrastructureResource(configv1.AzurePlatformType, configv1.AzureUSGovernmentCloud),
+		},
+		{
+			name:     "Azure keeps the cloud set to China cloud",
+			source:   azure.Config{AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzureChinaCloud)}},
+			expected: azure.Config{VMType: "standard", AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzureChinaCloud)}},
+			infra:    makeInfrastructureResource(configv1.AzurePlatformType, configv1.AzureChinaCloud),
+		},
+		{
+			name:     "Azure keeps the cloud set to German cloud",
+			source:   azure.Config{AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzureGermanCloud)}},
+			expected: azure.Config{VMType: "standard", AzureAuthConfig: ratelimitconfig.AzureAuthConfig{Cloud: string(configv1.AzureGermanCloud)}},
+			infra:    makeInfrastructureResource(configv1.AzurePlatformType, configv1.AzureGermanCloud),
+		},
+		// The matcher should assert err.Error == errMsg - but is not for some
+		// reason? According to:
+		// https://onsi.github.io/gomega/#matcherrorexpected-interface the matcher
+		// asserts that ACTUAL.Error() == EXPECTED
+		//	{
+		//		name:   "Azure throws an error if the infra has an invalid cloud",
+		//		source: azure.Config{},
+		//		infra:  makeInfrastructureResource(configv1.AzurePlatformType, "AzureAnotherCloud"),
+		//		errMsg: "status.platformStatus.azure.cloudName: Unsupported value: \"AzureAnotherCloud\": supported values: \"AzurePublicCloud\", \"AzureUSGovernmentCloud\", \"AzureChinaCloud\", \"AzureGermanCloud\", \"AzureStackCloud\""
+		//	},
```

Comment:
Was this copied or something you've written? Can you provide the failure output for when this is not commented out?

Reply:
It's something I've written. 

I think it is because we return a `field.NotSupported(` error. 

However, this should be matchable with the contents of it's `err.Error()`, hence my confusion.

EDIT: uncommenting and running `make test` locally this seems to be behaving as expected now 😵‍💫

Reply:
Wait no it's erroring again 😖 

```
=== RUN   TestCloudConfigTransformer/Azure_throws_an_error_if_the_infra_has_an_invalid_cloud


 err.Error():status.platformStatus.azure.cloudName: Unsupported value: "AzureAnotherCloud": supported values: "AzureUSGovernmentCloud", "AzureChinaCloud", "AzureGermanCloud", "AzureStackCloud", "AzurePublicCloud":

    azure_test.go:254:
        Expected
            <*field.Error | 0xc000069880>:
            status.platformStatus.azure.cloudName: Unsupported value: "AzureAnotherCloud": supported values: "AzureUSGovernmentCloud", "AzureChinaCloud", "AzureGermanCloud", "AzureStackCloud", "AzurePublicCloud"
            {
                Type: "FieldValueNotSupported",
                Field: "status.platformStatus.azure.cloudName",
                BadValue: <v1.AzureCloudEnvironment>"AzureAnotherCloud",
                Detail: "supported values: \"AzureUSGovernmentCloud\", \"AzureChinaCloud\", \"AzureGermanCloud\", \"AzureStackCloud\", \"AzurePublicCloud\"",
            }
        to match error
            <string>: status.platformStatus.azure.cloudName: Unsupported value: "AzureAnotherCloud": supported values: "AzurePublicCloud", "AzureUSGovernmentCloud", "AzureChinaCloud", "AzureGermanCloud", "AzureStackCloud"
```